File: src/main/java/it/tylframework/data/mongo/basics/Language.java
/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  * or more contributor license agreements.  See the NOTICE file
 *  * distributed with this work for additional information
 *  * regarding copyright ownership.  The ASF licenses this file
 *  * to you under the Apache License, Version 2.0 (the
 *  * "License"); you may not use this file except in compliance
 *  * with the License.  You may obtain a copy of the License at
 *  * <p/>
 *  * http://www.apache.org/licenses/LICENSE-2.0
 *  * <p/>
 *  * Unless required by applicable law or agreed to in writing,
 *  * software distributed under the License is distributed on an
 *  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  * KIND, either express or implied.  See the License for the
 *  * specific language governing permissions and limitations
 *  * under the License.
 */
package it.tylframework.data.mongo.basics;

import it.tylframework.data.mongo.common.Footprint;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * Created by mp on 20/11/14.
 */
@Document(collection="bas_language")
@TypeAlias("bas_language")
@Data
@RequiredArgsConstructor
public class Language extends Footprint{
    @NonNull String code;
    String flag;
    @NonNull String name;

}

File: src/main/java/it/tylframework/data/mongo/basics/Language.kt
package it.tylframework.data.mongo.basics

import it.tylframework.data.mongo.common.Footprint
import java.util.Date
import it.tylframework.data.mongo.common.Signature
import it.tylframework.data.mongo.config.TylContext
import org.springframework.data.annotation.TypeAlias

/**
 * Created by marco on 24/08/14.
 */
[TypeAlias("bas_Language")]
data class Language(
        var code: String,
        var flag: String,
        var name:String)
            :Footprint(TylContext.currentUser, Date() ,TylContext.currentUser,  Date()){
}
